### PR TITLE
make using venv in dockerfile

### PR DESCRIPTION
**Pull request by denieryd**
Now venv is used in dockerfile to set up dependencies, run app  

# Checklist

- [x] I've generated a new bot from the async-box template and checked that everything works:
  - [x] Linters and tests have passed
  - [x] Bot answers on `/help` command
  - [x] Bot suggests available commands
- [x] I've written a changelog for PR change
- [x] I'll make a release when PR is approved

### CODE REVIEW

File: {{cookiecutter.bot_project_name}}/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+ENV APP_USER=appuser
+ENV VIRTUAL_ENV=/home/$APP_USER/.venv/bin
+ENV USER_EXECUTABLES=/home/$APP_USER/.local/bin
+ENV PATH="$VIRTUAL_ENV:$USER_EXECUTABLES:$PATH"
 
-EXPOSE 8000
-WORKDIR /app
+RUN useradd --create-home $APP_USER
+WORKDIR /home/$APP_USER
+USER $APP_USER
```
Comment (vonabarak) [inline {{cookiecutter.bot_project_name}}/Dockerfile]: Is it possible to do `apt-get install` and `echo bla-bla-bla > /etc/sudoers.d/bla` without root privileges?

File: {{cookiecutter.bot_project_name}}/Dockerfile [hunk 2]
```diff
@@ -20,11 +30,9 @@ ARG GIT_LOGIN="gitlab-ci-token"
 RUN echo -e "machine ${GIT_HOST}\nlogin ${GIT_LOGIN}\npassword ${GIT_PASSWORD}" > ~/.netrc && \
     apt-get update && \
     apt-get install -y git sudo && \
-    pip install poetry==1.1.6 --no-cache-dir && \
-    poetry config virtualenvs.create false && \
     poetry install --no-dev && \
-    echo "${NON_ROOT_USER} ALL = NOPASSWD: /usr/sbin/update-ca-certificates" > /etc/sudoers.d/express_bot && \
-    rm -rf /root/.cache/pypoetry && \
+    echo "${APP_USER} ALL = NOPASSWD: /usr/sbin/update-ca-certificates" > /etc/sudoers.d/${APP_USER} && \
```
Comment (vonabarak) [inline {{cookiecutter.bot_project_name}}/Dockerfile]: There is no need to name a file in `/etc/sudoers.d/` same as username
But maybe name `express_bot` really not very descriptive. I suggest naming it `/etc/sudoers.d/update-ca-certificates` or something like that.

File: {{cookiecutter.bot_project_name}}/Dockerfile [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+EXPOSE 8000
+
 ENV PYTHONUNBUFFERED 1
 ENV UVICORN_CMD_ARGS ""
 
-ARG NON_ROOT_USER=express_bot
-RUN useradd --create-home ${NON_ROOT_USER}
+ENV APP_USER=appuser
+ENV VIRTUAL_ENV=/home/$APP_USER/.venv/bin
+ENV USER_EXECUTABLES=/home/$APP_USER/.local/bin
```
Comment (vonabarak) [inline {{cookiecutter.bot_project_name}}/Dockerfile]: I suggest rename VIRTUAL_ENV to VENV_PATH (or mb PATH_VENV) and USER_EXECUTABLES to USER_PATH (or PATH_USER)

File: {{cookiecutter.bot_project_name}}/Dockerfile (hunk 1 above)
Comment (denieryd) [reply {{cookiecutter.bot_project_name}}/Dockerfile]: Hm, you're right, thanks. Changed the order of commands to the right.

File: {{cookiecutter.bot_project_name}}/Dockerfile (hunk 2 above)
Comment (denieryd) [reply {{cookiecutter.bot_project_name}}/Dockerfile]: Changed

File: {{cookiecutter.bot_project_name}}/Dockerfile (hunk 3 above)
Comment (denieryd) [reply {{cookiecutter.bot_project_name}}/Dockerfile]: Changed

**Review by vonabarak — APPROVED** [review]

File: {{cookiecutter.bot_project_name}}/Dockerfile [hunk 4]
```diff
@@ -1,43 +1,50 @@
...
 RUN echo -e "machine ${GIT_HOST}\nlogin ${GIT_LOGIN}\npassword ${GIT_PASSWORD}" > ~/.netrc && \
     apt-get update && \
     apt-get install -y git sudo && \
-    pip install poetry==1.1.6 --no-cache-dir && \
-    poetry config virtualenvs.create false && \
-    poetry install --no-dev && \
-    echo "${NON_ROOT_USER} ALL = NOPASSWD: /usr/sbin/update-ca-certificates" > /etc/sudoers.d/express_bot && \
-    rm -rf /root/.cache/pypoetry && \
+    echo "${APP_USER} ALL = NOPASSWD: /usr/sbin/update-ca-certificates" > /etc/sudoers.d/update-ca-certificates && \
+    sudo update-ca-certificates && \
```
Comment (vonabarak) [inline {{cookiecutter.bot_project_name}}/Dockerfile]: Unnecessary sudo.

**Review by Opti213 — APPROVED** [review]

File: {{cookiecutter.bot_project_name}}/Dockerfile [hunk 5]
```diff
@@ -1,43 +1,50 @@
...
 
 ARG CI_COMMIT_SHA=""
 ENV GIT_COMMIT_SHA=${CI_COMMIT_SHA}
 
-USER ${NON_ROOT_USER}
-
-CMD sudo update-ca-certificates && \
-    alembic upgrade head && \
-    uvicorn --host=0.0.0.0 $UVICORN_CMD_ARGS app.main:app
+CMD alembic upgrade head && \
```
Comment (vonabarak) [inline {{cookiecutter.bot_project_name}}/Dockerfile:49]: Why update-ca-certificates removed?

File: {{cookiecutter.bot_project_name}}/Dockerfile [hunk 6]
```diff
@@ -1,43 +1,50 @@
...
     apt-get update && \
     apt-get install -y git sudo && \
-    pip install poetry==1.1.6 --no-cache-dir && \
-    poetry config virtualenvs.create false && \
-    poetry install --no-dev && \
-    echo "${NON_ROOT_USER} ALL = NOPASSWD: /usr/sbin/update-ca-certificates" > /etc/sudoers.d/express_bot && \
-    rm -rf /root/.cache/pypoetry && \
+    echo "${APP_USER} ALL = NOPASSWD: /usr/sbin/update-ca-certificates" > /etc/sudoers.d/update-ca-certificates && \
+    sudo update-ca-certificates && \
+    rm -rf ~/.cache/pypoetry && \
```
Comment (vonabarak) [inline {{cookiecutter.bot_project_name}}/Dockerfile]: This command should run after `poetry install`, not before.

File: {{cookiecutter.bot_project_name}}/Dockerfile [hunk 7]
```diff
@@ -1,43 +1,50 @@
...
+RUN useradd --create-home $APP_USER
+WORKDIR /home/$APP_USER
+USER $APP_USER
+
+ENV VENV_PATH=/home/$APP_USER/.venv/bin
+ENV USER_PATH=/home/$APP_USER/.local/bin
+ENV PATH="$VENV_PATH:$USER_PATH:$PATH"
+
+RUN pip install --user --no-cache-dir poetry==1.1.6 && \
+    poetry config virtualenvs.in-project true
```
Comment (vonabarak) [inline {{cookiecutter.bot_project_name}}/Dockerfile]: `poetry config virtualenvs.in-project true` makes poetry to create venv in project directory, while PATH assumes venv in user's home.

**Review by vonabarak — APPROVED** [review]

**Pull request merged**